src/sinks/gcp_chronicle/partitioner.rs

```diff
@@ -9,11 +9,23 @@ pub struct ChroniclePartitionKey {
 }
 
 /// Partitions items based on the generated key for the given event.
-pub struct ChroniclePartitioner(Template, Option<Template>);
+pub struct ChroniclePartitioner {
+    log_type: Template,
+    template_fallback_log_type: Option<String>,
```

Comment:
Nit: 
```suggestion
    fallback_log_type: Option<String>,
```

Is this intentionally a `String` and not a `Template`?

Reply:
Yea, it's the fallback option if the `log_type` template can't be resolved, similar to [this Elasticsearch change](https://github.com/vectordotdev/vector/pull/21953)